src/contour/TerminalWindow.cpp

```diff
@@ -107,6 +130,15 @@ TerminalWindow::TerminalWindow(std::chrono::seconds _earlyExitThreshold,
     if (config_.maxImageSize.height <= Height(0))
         config_.maxImageSize.height = defaultMaxImageSize.height;
     // }}}
+    if (config_.profile().sessionResume)
+    {
+        auto [configPath, profile, gridBuffer] = loadSessionFile(crispy::xdgStateHome() / "contour/session");
```

Comment:
What if you had 3 terminal windows open, log out and back in again, then session restore kicks back in. Are all 3 windows being restored properly?

I think what is of interest is:

- https://doc.qt.io/qt-5/session.html
- https://doc.qt.io/qt-5/qguiapplication.html#sessionId

So if for example you're running a KDE desktop (not sure about other Linux desktops, never tried), and you had 3 contour processes open, they're all going to be respawned again by KDE desktop. Each having a session id passed via CLI (Contour actually properly passes this to underlying Qt's Application). 
My thinking is, that passed session Id could then be used to properly handle multiple stored sessions.